TypeCobol/Compiler/CompilationUnit.cs

```diff
@@ -35,8 +34,16 @@ public class CompilationUnit : CompilationDocument
             PerfStatsForTemporarySemantic = new PerfStatsForParsingStep(CompilationStep.ProgramClassParser);
             PerfStatsForProgramCrossCheck = new PerfStatsForParsingStep(CompilationStep.ProgramCrossCheck);
             PerfStatsForCodeQualityCheck = new PerfStatsForCompilationStep(CompilationStep.CodeQualityCheck);
+            if (analyzerProvider != null)
+            {
+                if (!(analyzerProvider is AnalyzerProviderWrapper analyzerProviderWrapper))
+                {
+                    analyzerProviderWrapper = new AnalyzerProviderWrapper();
+                    analyzerProviderWrapper.AddProvider(analyzerProvider);
+                }
+                _analyzerProvider = analyzerProviderWrapper;
+            }
```

Comment:
It's correct but I think it would be clearer that way:
```csharp
if (analyzerProvider is AnalyzerProviderWrapper analyzerProviderWrapper)
{
    _analyzerProvider = analyzerProviderWrapper;
}
else if (analyzerProvider != null)
{
    _analyzerProvider = new AnalyzerProviderWrapper();
    _analyzerProvider.AddProvider(analyzerProvider);
}
else
{
    _analyzerProvider = null;
}
```

It's more verbose but it makes the possible values for the field more apparent.

In fact I wonder if we should allow `null` for the field. An empty wrapper could be used.

Reply:
The initialization is now more verbose.
Concerning allowing or not the value `null`, I think keeping it `null` stays in the spirit of the parser which is to have null instead of empty object.